tests/linearizability/model.go

```diff
@@ -99,22 +98,32 @@ func step(state EtcdState, request etcdRequest, response etcdResponse) (bool, Et
 	switch request.op {
 	case Get:
 		if state.Value == response.getData {
-			return true, state
+			if state.LastRevision == 0 {
+				state.LastRevision = response.revision
+			}
+			if state.LastRevision == response.revision {
+				return true, state
+			}
 		}
-		for write := range state.FailedWrites {
-			if write == response.getData {
+		if state.LastRevision < response.revision {
+			_, ok := state.FailedWrites[response.getData]
+			if ok {
 				state.Value = response.getData
-				delete(state.FailedWrites, write)
+				state.LastRevision = response.revision
+				delete(state.FailedWrites, response.getData)
 				return true, state
 			}
 		}
```

Comment:
It's a little hard to understand. I would suggest to rewrite the case Get branch something like below,

```
		if state.LastRevision == 0 {
			// run into this branch if there is no any previous successful
			// put operations.
			state.Value = response.getData
			state.LastRevision = response.revision
			return true, state
		}
		if state.Value == response.getData && state.LastRevision == response.revision {
			return true, state
		} else {
			_, ok := state.FailedWrites[response.getData]
			if ok {
				state.Value = response.getData
				state.LastRevision = response.revision
				delete(state.FailedWrites, response.getData)
				return true, state
			}
		}
```